os/tools/mkheader.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+############################################################################
+#
+# Copyright 2019 Samsung Electronics All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+############################################################################
+import os
+import sys
+import struct
+############################################################################
+#
+# header information :
+#
+# / header size (2byte) / binary type (2byte) / binary size (4byte)
+# / Binary name (16byte) / Binary version (16byte)
+# / Kernel version (8byte) / Jump address (4byte)
+# header size is 52byte.
+#
+# parameter information :
+#
+# argv[1] is file path of binary file.
+# argv[2] is file type.
+# argv[3] is kernel version.
+# argv[4] is binary name.
+# argv[5] is binary version.
+#
+############################################################################
+
+header_size = 52
+file_path  = sys.argv[1]
+binary_type = sys.argv[2]
+kernel_ver = sys.argv[3]
+binary_name = sys.argv[4]
+binary_ver = sys.argv[5]
+
+tinyara_path = 'root/tizenrt/build/output/bin/tinyara'
+
+START = 0
+HEADER_SIZE_S = 2
+BINARY_TYPE_S = 2
+BINARY_SIZE_S = 4
+BINARY_NAME_S = 16
+BINARY_VERSION_S = 16
+KERNEL_VERSION_S = 8
+JUMP_ADDRESS_S = 4
+
+ELF = 1
+KERNEL = 2
+
+with open(file_path, 'rb') as fp:
+    # binary data copy to 'data'
+    data = fp.read()
+    file_size = fp.tell()
+    fp.close()
+    fp = open(file_path, 'wb')
+
+    # header initialization
+    for i in range(0, header_size):
+        fp.write('\000')
+
+    OFFSET = START
+
+    fp.seek(OFFSET,0)
+    fp.write(struct.pack('h', header_size))
+    OFFSET += HEADER_SIZE_S
+
+    if binary_type == 'kernel' or binary_type == 'KERNEL' :
```

Comment:
I think 'kernel' and 'elf' is not in the same hierarchy.
how about 'bin' and 'elf'?